### DISCUSSION TITLE
InspireHEP badges for publications

### DISCUSSION THREAD

**Discussion by riasat-sheikh** [alshedivat/al-folio#2634] [Q&A] [answered]
Is there any way to get the InspireHEP citation count for an article in my publication list similar to the google scholar badge. 

If you click the citation button of an article in the InspireHEP page it will search for something like refersto:recid:{article_ID}. Maybe there is some way?

**Comment by george-gca** [comment, marked as answer]
Probably yes (they have a [documented API](https://github.com/inspirehep/rest-api-doc)), but you have to implement it yourself since they don't have badges like [altmetric](https://www.altmetric.com/solutions/altmetric-badges/). You probably have to retrieve information about the paper using this API, parse the result, then show it somehow.

For displaying is probably useful to do how it was [done for Google Scholar](https://github.com/alshedivat/al-folio/blob/3ff7579a7419fc546816535361a8b90c7c49553d/_layouts/bib.liquid#L309) using [shields](https://shields.io/).

**Reply by riasat-sheikh** [reply]
Thank you very much for your response.

I followed your suggestion and successfully added it to [my website](https://thefermi0n.github.io/).

By using the [google-scholar-citation.rb](https://github.com/alshedivat/al-folio/blob/3ff7579a7419fc546816535361a8b90c7c49553d/_plugins/google-scholar-citations.rb) plugin as a reference, I wrote a similar plugin for Inspire HEP and modified the bib.liquid accordingly.

Since the Inspire API documentation didn't mention any time delay, I retained the same randomized delay from the Google Scholar plugin.

Should I create a pull request to add this to the al-folio template?

**Reply by george-gca** [reply]
I believe that others might benefit from this, so yes, please.
